### DISCUSSION TITLE
Trouble with Communication Configuration Updates in MPyC: New Configuration Not Taking Effect

### DISCUSSION THREAD

**Discussion by qqqqqqQwQ** [lschoe/mpyc#87] [Q&A] [answered]
I have a question about communication configuration: I created a new folder at E:\desktop\mpc and then copied unanimous.py, .config/generate-config-files.py, and generate-certificates.py into it. I used "python generate-config-files.py -m 3 :11111 :11112 :11113" and "python generate-certificates.py -m 3" to regenerate the configuration. I added print(mpc.parties[0:]) to unanimous.py. However, when I execute "python unanimous.py -M3 -I0 1", the terminal displays "(<Party 0: localhost:11365>, <Party 1: localhost:11366>, <Party 2: localhost:11367>)", which is the original ip:port configuration from the demos. Even after restarting, the new configuration still does not take effect. How can I solve this?

**Comment by lschoe** [comment, marked as answer]
Using `python unanimous.py -H` or `python -m mpyc -H` you can inspect all [MPyC command line options](https://mpyc.readthedocs.io/en/latest/cli.html) and see how to do this.

If you want to work with config files, this option applies: 

`  -C ini, --config ini  use ini file, defining all m parties`

You can get the same result for your case without config files, by combining the use of these three options:

`  -M m                  use m local parties (and run all m, if i is not set)`
`  -I i, --index i       set index of this local party to i, 0<=i<m`
` -B b, --base-port b   use port number b+i for party i`

Like this on three terminals:

`$ python unanimous.py -B11111 -M3 -I2 --log debug`
`$ python unanimous.py -B11111 -M3 -I1 --log debug`
`$ python unanimous.py -B11111 -M3 -I0 --log debug`

By setting the log-level to debug, you'll also see which ports are used.

**Reply by qqqqqqQwQ** [reply]
Thank you for your answer!
